lib/datasource.js

```diff
@@ -327,13 +327,22 @@ DataSource.prototype.setup = function (name, settings) {
 
     }.bind(this);
 
-    if ('function' === typeof connector.initialize) {
-      // Call the async initialize method
-      connector.initialize(this, postInit);
-    } else if ('function' === typeof connector) {
-      // Use the connector constructor directly
-      this.connector = new connector(this.settings);
-      postInit();
+    try {
+      if ('function' === typeof connector.initialize) {
+        // Call the async initialize method
+        connector.initialize(this, postInit);
```

Comment:
> we should remove the stupid initialize faux constructor at some point

IIRC, all our connectors use `connector.initialize` instead of a class constructor :(

/cc @superkhau perhaps this is something you can include in your refactoring of connectors?